code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,270 @@
+				P.owner = owner
+				var/list/owners = P.get_owners()
+				P.find_target(owners,assasination)
+				protection += P.target
+				objectives += P
+
+	var/datum/objective/sacrifice_ecult/SE = new
+	SE.update_explanation_text()
+	objectives += SE
+
+	return
+
+/datum/antagonist/ecult/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, mob_override ? null : "Knowledge described in the book allowed you to overcome your clownish nature, allowing you to use complex items effectively.")
+	current.faction |= "ecult"
+
+/datum/antagonist/ecult/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, removing = FALSE)
+	current.faction -= "ecult"
+
+/datum/antagonist/ecult/roundend_report()
+	var/list/parts = list()
+
+	var/cultiewin = TRUE
+
+	parts += printplayer(owner)
+
+	//Removed sanity if(changeling) because we -want- a runtime to inform us that the changelings list is incorrect and needs to be fixed.
+	parts += "<b>Sacrifices Made:</b> [total_sacrifices]"
+
+	if(objectives.len)
+		var/count = 1
+		for(var/datum/objective/objective in objectives)
+			if(objective.check_completion())
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='greentext'>Success!</b></span>"
+			else
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='redtext'>Fail.</span>"
+				cultiewin = FALSE
+			count++
+
+	if(cultiewin)
+		parts += "<span class='greentext'>The eldritch cultist was successful!</span>"
+	else
+		parts += "<span class='redtext'>The eldritch cultist has failed.</span>"
+
+	parts += "<b>Knowledge Researched:</b> "
+
+	var/knowledge_message = ""
+	for(var/X in get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		knowledge_message += "[EK.name], "
+	parts += knowledge_message
+
+	return parts.Join("<br>")
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/ecult/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge |= initialized_knowledge
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/ecult/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/ecult/proc/has_knowledge(datum/eldritch_knowledge/EK)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK1 = X
+		if(initial(EK.name) == EK1.name)
```

Comment:
Please use more informative names than `EK` and `EK1`
You could try things like `eldritch_knowledge_to_check` and `researched_eldritch_knowledge`